website/www/site/content/en/documentation/programming-guide.md

```diff
@@ -1184,6 +1184,101 @@ func init() {
 
 </span>
 
+{{< paragraph class="language-python">}}
+Proper Use of return vs yield in Python Functions.
+{{< /paragraph >}}
+
+{{< highlight python >}}
+# Returning a single string instead of a sequence
+class ReturnIndividualElement(beam.DoFn):
+    def process(self, element):
+        return element  
```

Comment:
I have run into this problem while working on stateful processing. I would also update those programming guide to explicitly refer this section. Otherwise, people spending a lot of time debugging their code. 

Ideally, python typecheck linter should error out if the type information is set incorrectly.